pkg/kubelet/status/status_manager.go

```diff
@@ -879,9 +887,26 @@ func mergePodStatus(oldPodStatus, newPodStatus v1.PodStatus, couldHaveRunningCon
 			podConditions = append(podConditions, c)
 		}
 	}
+
+	transitioningToTerminalPhase := !podutil.IsPodPhaseTerminal(oldPodStatus.Phase) && podutil.IsPodPhaseTerminal(newPodStatus.Phase)
+
 	for _, c := range newPodStatus.Conditions {
 		if kubetypes.PodConditionByKubelet(c.Type) {
 			podConditions = append(podConditions, c)
+		} else if kubetypes.PodConditionSharedByKubelet(c.Type) {
```

Comment:
Thanks for updating the logic like this -- this is much cleaner than the previous approach of "reverting" the condition. 

I think it's much more clear now: we replace/append all the "shared kubelet" conditions, and guard the `AlphaNoCompatGuaranteeDisruptionTarget` condition with `transitioningToTerminalPhase && !couldHaveRunningContainers` to ensure it will only be sent once all containers have terminated.

Reply:
Thanks for point this out with the details. 

nit: Can we include above the logic as the comment for future maintainence?

Reply:
Added a comment in the follow up PR: https://github.com/kubernetes/kubernetes/pull/113718